src/config.cr

```diff
@@ -1,20 +1,21 @@
 require "./access"
 require "./ext/scrypt_password"
+require "json"
 
 struct DppmRestApi::Config
   include JSON::Serializable
   property groups : Array(Group)
   property users : Array(User)
 
-  def user(*, named username : String) : User?
+  def user_from_name(username : String) : User?
     users.find { |user| user.name == username }
   end
 
-  def user(*, authenticated_with api_key : String)
+  def user_authenticated_by(api_key : String)
```

Comment:
If we have a stack trace, with `user_authenticated_by` won't give us any clue what is it.
```suggestion
  def authenticated_user_from_api_key(api_key : String)
```
A bit long, but is it really a problem? Not sure we will use it that often.

Reply:
It's not that it was a problem I just had done that before you suggested your version. I'll change it now.

Reply:
Actually, the *only* place where it even *would* be used... it [isn't](https://github.com/DFabric/dppm-rest-api/blob/b8b514f40038009c177c5dab8279bce21f41a339/src/middlewares/auth_middleware.cr#L34). I think we should just remove these methods.

Reply:
So I realize it's not working right now, but I wanted to get the idea fleshed out and see what you think. I think my Kemal Auth library can actually be reworked to be simpler, making use of modules as interfaces rather than needing a generic to be defined. I think I'm going to look into that before going on.